### PR TITLE
Simplify DYE mod

### PR DESCRIPTION
**Pull request by tenplus1**
This simplifies the dye mod and removes the fluff that is no longer being used.

EDIT by paramat: To be merged with #2142

### CODE REVIEW

**Comment by tenplus1** [comment]
@paramat - sorry I had to re-add this pull as the previous one had issues with me editing it.

**Comment by paramat** [comment]
Why is 'unicolor' kept here but removed for wools? I seems to remember it's due to someone's popular mod, but seems inconsistent.

**Comment by tenplus1** [comment]
This greatly simplifies the dye groups and crafting and the only reason for unicolor is backwards compatibility with vanessa's mods that use paramtype colouring which cannot be simplified.

**Comment by VanessaE** [comment]
I'm perfectly happy to make the required updates to my mods if needed, though none of my mods appear to use any of those groups.

**Comment by paramat** [comment]
tenplus1 could you explain why unicolor is left in here?

**Comment by tenplus1** [comment]
@VanessaE - https://gitlab.com/VanessaE/stained_glass/blob/master/init.lua#L292

**Comment by VanessaE** [comment]
@tenplus1 that's just a fork, as I'm not the maintainer, but anyway, ignore that usage.  The mod uses the Unified Dyes place-and-punch-with-dye paradigm anyway, so the craft recipes won't work (maybe I'll fix that soon); "breaking" them with this PR or any other will have no effect. :smiley:

See also https://github.com/andersje/stained_glass/commits/master (cloeset one "maintainer" I can think of, or https://github.com/doyousketch2/stained_glass/commits/master (the original author, badly outdated).

**Comment by paramat** [comment]
Good news, nevermind explaining then. Please could this PR be reverted to the initial form? then i will support both.

**Comment by tenplus1** [comment]
two were found, one already patched, the other stained glass.

**Comment by paramat** [comment]
Ok no problem.

File: mods/dye/init.lua [hunk 1]
```diff
@@ -1,9 +1,5 @@
--- Other mods can use these for looping through available colors
```
Comment (paramat) [inline mods/dye/init.lua]: Unnecessary empty line

File: mods/dye/init.lua [hunk 2]
```diff
@@ -25,47 +21,30 @@ dye.dyes = {
...
 	local description = row[2]
-	local groups = row[3]
-	local item_name = "dye:" .. name
-	local item_image = "dye_" .. name .. ".png"
-	minetest.register_craftitem(item_name, {
-		inventory_image = item_image,
-		description = description,
+
+	-- Add groups
+	local groups = {dye = 1, flammable = 2}
```
Comment (paramat) [inline mods/dye/init.lua]: Please remove 'flammable'

File: mods/dye/init.lua [hunk 3]
```diff
@@ -25,47 +21,30 @@ dye.dyes = {
...
 	local name = row[1]
 	local description = row[2]
-	local groups = row[3]
-	local item_name = "dye:" .. name
-	local item_image = "dye_" .. name .. ".png"
-	minetest.register_craftitem(item_name, {
-		inventory_image = item_image,
-		description = description,
+
+	-- Add groups
```
Comment (paramat) [inline mods/dye/init.lua]: No need for this comment or the empty line above

**Comment by paramat** [comment]
+1 once line comments attended to.

**Comment by tenplus1** [comment]
Why remove flammable group ? don't you want it to burn up if dropped inside fire or lava ?

**Comment by paramat** [comment]
Unsure about flammable, because dye essentially isn't. But we can consider that later.
Line 1 is empty and still needs deleting.

**Comment by paramat** [comment]
:+1: and assuming +1 from Ezhh

**Pull request merged**